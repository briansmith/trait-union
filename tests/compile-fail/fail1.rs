use trait_union::trait_union;
use std::fmt::Debug;

trait_union! {
    union U<'a>: Debug = &'a str;
}

fn main() { }
