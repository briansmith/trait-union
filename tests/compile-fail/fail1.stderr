error[E0478]: lifetime bound not satisfied
 --> $DIR/fail1.rs:5:11
  |
5 |     union U<'a>: Debug = &'a str;
  |           ^^^^^
  |
note: lifetime parameter instantiated with the lifetime `'a` as defined on the impl at 5:13
 --> $DIR/fail1.rs:5:13
  |
5 |     union U<'a>: Debug = &'a str;
  |             ^^
  = note: but lifetime parameter must outlive the static lifetime
